packages/mask/src/plugins/Tips/components/Guide.tsx

```diff
@@ -0,0 +1,194 @@
+            display: 'inline-block',
+            width: 0,
+            height: 0,
+            border: 'solid 8px transparent',
+            borderBottomColor: theme.palette.maskColor.tips,
+            borderBottomWidth: '13px',
+            borderTopWidth: 0,
+            position: 'absolute',
+            top: '-13px',
+            left: '24px',
+        },
+        '&.arrow-bottom:after': {
+            content: '""',
+            display: 'inline-block',
+            width: 0,
+            height: 0,
+            border: 'solid 8px transparent',
+            borderTopColor: theme.palette.maskColor.tips,
+            borderTopWidth: '13px',
+            borderBottomWidth: 0,
+            position: 'absolute',
+            bottom: '-13px',
+            left: '24px',
+        },
+    },
+    buttonContainer: {
+        display: 'flex',
+        justifyContent: 'space-between',
+        paddingTop: '16px',
+    },
+    button: {
+        width: '100%',
+        borderRadius: '20px',
+    },
+}))
+
+export interface GuideStepProps extends PropsWithChildren<{}> {
+    arrow?: boolean
+    disabled?: boolean
+    onComplete?: () => void
+}
+
+export default function Guide({ children, arrow = true, disabled = false, onComplete }: GuideStepProps) {
+    const t = useI18N()
+    const { classes, cx } = useStyles()
+    const childrenRef = useRef<HTMLElement>()
+    const [clientRect, setClientRect] = useState<any>({})
+    const [open, setOpen] = useState(false)
+    const [bottomAvailable, setBottomAvailable] = useState(true)
+    const ui = activatedSocialNetworkUI
+    const enableUserGuide = ui.configuration.tipsConfig?.enableUserGuide
+    const lastStep = useTipsUserGuide(ui.networkIdentifier as EnhanceableSite)
+
+    useEffect(() => {
+        if (disabled || !enableUserGuide || lastStep.finished) return
+        setOpen(true)
+    }, [])
+
+    useEffect(() => {
+        if (disabled || !enableUserGuide || lastStep.finished) return
+        document.body.style.overflow = open ? 'hidden' : ''
+    }, [open])
+
+    const onNext = () => {
+        setOpen(false)
+        finishUserGuide(ui.networkIdentifier as EnhanceableSite)
+    }
+
+    useEffect(() => {
+        if (disabled || !enableUserGuide || lastStep.finished) return
+        const onResize = () => {
+            const cr = childrenRef.current?.getBoundingClientRect()
+
+            if (cr) {
+                const bottomAvailable = window.innerHeight - cr.height - cr.top > 200
+                setBottomAvailable(bottomAvailable)
+                if (!cr.width) {
+                    setClientRect({ ...cr, top: 30, left: 'calc(100vw - 300px)' })
+                } else {
+                    setClientRect(cr)
+                }
+            }
+        }
+
+        onResize()
+
+        window.addEventListener('resize', onResize)
+
+        return () => {
+            window.removeEventListener('resize', onResize)
+        }
+    }, [childrenRef, lastStep.finished])
```

Comment:
Lost deps here.

Reply:
Yep, I think the `disabled` and `enableUserGuide` not be change in runtime.